_posts/2020-10-28-push-it-to-the-limit.md

```diff
@@ -0,0 +1,58 @@
+---
+layout: post
+title: Push It To The Limit
+categories:
+- blog
+tags:
+- MUD
+- Open Source
+- Optimization
+- Firebase
+- Lich
+---
+
+I've been hosting portions of the Lich framework I help maintain on [Firebase](https://firebase.google.com/) using a free tier plan for several years now. I became inured to the monthly e-mails reminding me that I was at 90% of my data limit showing up on the ~~30th~~ ~~29th~~ ~~28th~~... shit.
+
+---
+
+## Punished By Success
+
+It was probably inevitable that we would reach this point, adoption of our project had been growing steadily and user behavior was changing in ways that increased their data demands. I wasn't going to start giving google nickles of my money each month without a fight so I prepared to spend hours of my time preventing this tragedy. To provide more context I will first answer the obvious question, why does a MUD script repository need a database?
```

Comment:
> google nickles of my money each month without a fight so I prepared to spend hours of my time preventing this tragedy

This reminds me of the old joke "a few hours of trial and error can save you several minutes of looking at the README"